### PR TITLE
Check if the module exists on Addons

### LINKED ISSUE #18499: Theme installation error during silent installation
#### Describe the bug

Silent installation fail when using the option --theme

#### Expected behavior

The installation succeed and the selected theme is activated

#### Steps to Reproduce
1 set-up a proper environment to install PrestaShop (mysql, apache,...)
2 unzip prestashop 1.7.6.4
3 unzip [classic-rocket](https://github.com/PrestaShop/classic-rocket/releases)  into themes/classic-rocket
4 launch the silent installation
php install/index_cli.php --language=en --country=fr --domain=test.localtest.me --db_server=localhost --db_user=presta  --db_password=presta --db_name=test --firstname=John --lastname=doe --email=test@test.com --password=azerty01 --newsletter=0 --theme=classic-rocket

Result:
The module is invalid and cannot be loaded.

#### Additional information
After a successful installation if you upload classic-rocket throught the web form it works.

But if you unzip it directy in the folder and then click on "use this theme" in the web interface it fails with this error message: "Cannot install module ps_searchbarjqauto. The module is invalid and cannot be loaded."

* PrestaShop version: 1.7.6.4
* PHP version: 7.2

#### Bonus
Documentation is outdated:
http://doc.prestashop.com/display/PS17/Installing+PrestaShop+using+the+command-line+script

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Before this PR, the CLI trigger a mystic error when a module was not found on Addons
| Type?         | improvement
| Category?     | IN
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #18499
| How to test?  | 1. set-up a proper environment to install PrestaShop (mysql, apache,...)<br>2. work on `develop` branch<br>3. unzip [classic-rocket](https://github.com/PrestaShop/classic-rocket/releases)  into themes/classic-rocket<br>4. launch the silent installation : `php install-dev/index_cli.php --language=en --country=fr --domaine='127.0.0.1' --base_uri='/PrestaShop/' --db_server=127.0.0.1 --db_user=root --db_password=root --db_name=prestashop --step=database,theme,modules,addons_modules --theme=classic-rocket`<br>5. **BEFORE**: "Cannot install module ps_searchbarjqauto. The module is invalid and cannot be loaded."<br>5. **AFTER**: "Module ps_searchbarjqauto not found on Addons."

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19744)

### CODE REVIEW

**Review by NeOMakinG — DISMISSED** [review]

File: src/Core/Addon/Module/ModuleManager.php [hunk 1]
```diff
@@ -622,6 +624,7 @@ public function getError($name)
             $errors = $module->getInstance()->getErrors();
             $message = array_pop($errors);
         } else {
+            debug_print_backtrace();
```
Comment (PierreRambaud) [inline src/Core/Addon/Module/ModuleManager.php]: Should only be done in CLI :scream:

**Review by PierreRambaud — DISMISSED** [review]
Avoid `debug_print_backtrace()` without any context :/

File: src/Core/Addon/Module/ModuleManager.php (hunk 1 above)
Comment (matks) [reply src/Core/Addon/Module/ModuleManager.php]: YOLO

Comment (Progi1984) [reply src/Core/Addon/Module/ModuleManager.php]: Oops.... I go out :p

Comment (Progi1984) [reply src/Core/Addon/Module/ModuleManager.php]: Fixed

**Review by atomiix — DISMISSED** [review]

File: src/Core/Addon/Module/ModuleManager.php [hunk 2]
```diff
@@ -294,7 +294,9 @@ public function install($source)
         if (!empty($source)) {
             $this->moduleZipManager->storeInModulesFolder($source);
         } elseif (!$this->moduleProvider->isOnDisk($name)) {
-            $this->moduleUpdater->setModuleOnDiskFromAddons($name);
+            if (!$this->moduleUpdater->setModuleOnDiskFromAddons($name)) {
+                throw new Exception($this->translator->trans('The module %name% has not been found on Addons.', ['%name%' => $name]));
```
Comment (Quetzacoalt91) [inline src/Core/Addon/Module/ModuleManager.php]: What domain is taken here if you do not specify 'Admin.Modules.Notification' ?

Comment (Progi1984) [reply src/Core/Addon/Module/ModuleManager.php]: I'm waiting Team Wording ;). I don't find any correct wording.

Comment (Junebyun) [inline src/Core/Addon/Module/ModuleManager.php]:
```suggestion
                throw new Exception($this->translator->trans('The module %name% could not be found on Addons.', ['%name%' => $name])); 'Admin.Modules.Notification'
```

Comment (Quetzacoalt91) [reply src/Core/Addon/Module/ModuleManager.php]: Be careful, there's a syntax error in your suggestion.

```suggestion
                throw new Exception($this->translator->trans('The module %name% could not be found on Addons.', ['%name%' => $name], 'Admin.Modules.Notification'));
```

**Review by Junebyun — DISMISSED** [review]

**Comment by Junebyun** [comment]
@LouiseBonnard wording approved :)

**Review by LouiseBonnard — DISMISSED** [review]

**Comment by PierreRambaud** [comment]
I'm wondering, what if you try to install the same theme with the BO interface?


From the original issue:
```
Additional information
After a successful installation if you upload classic-rocket throught the web form it works.

But if you unzip it directy in the folder and then click on "use this theme" in the web interface it fails with this error message: "Cannot install module ps_searchbarjqauto. The module is invalid and cannot be loaded."

```

**Comment by Progi1984** [comment]
> I'm wondering, what if you try to install the same theme with the BO interface?
> 
> From the original issue:
> 
> ```
> Additional information
> After a successful installation if you upload classic-rocket throught the web form it works.
> 
> But if you unzip it directy in the folder and then click on "use this theme" in the web interface it fails with this error message: "Cannot install module ps_searchbarjqauto. The module is invalid and cannot be loaded."
> ```

Good idea. I fix it too.

**Comment by Progi1984** [comment]
@PierreRambaud Fixed

**Review by PierreRambaud — APPROVED** [review]

**Review by LouiseBonnard — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by Junebyun — APPROVED** [review]

**Comment by SD1982** [comment]
LGTM Thanks @Progi1984 !!

**Comment by Progi1984** [comment]
Thanks @SD1982

**Pull request merged**